finality-aleph/src/network.rs

```diff
@@ -24,48 +59,50 @@ pub trait Network<B: BlockT>: Clone + Send + Sync + 'static {
     fn event_stream(&self) -> Pin<Box<dyn Stream<Item = Event> + Send>>;
 
     /// Adjust the reputation of a node.
-    fn _report_peer(&self, peer_id: PeerId, reputation: ReputationChange);
+    fn _report_peer(&self, peer_id: ScPeerId, reputation: ReputationChange);
 
     /// Force-disconnect a peer.
-    fn _disconnect_peer(&self, peer_id: PeerId, protocol: Cow<'static, str>);
+    fn _disconnect_peer(&self, peer_id: ScPeerId, protocol: Cow<'static, str>);
 
     /// Send a message to a given peer.
-    fn send_message(&self, peer_id: PeerId, protocol: Cow<'static, str>, message: Vec<u8>);
+    fn send_message(&self, peer_id: ScPeerId, protocol: Cow<'static, str>, message: Vec<u8>);
 
     /// Notify everyone we're connected to that we have the given block.
     /// This might be useful in the future.
     fn _announce(&self, block: B::Hash, associated_data: Option<Vec<u8>>);
 
     /// TODO: figure out what does this actually do...
-    fn add_set_reserved(&self, who: PeerId, protocol: Cow<'static, str>);
+    fn add_set_reserved(&self, who: ScPeerId, protocol: Cow<'static, str>);
 
     /// TODO: figure out what does this actually do...
-    fn remove_set_reserved(&self, who: PeerId, protocol: Cow<'static, str>);
+    fn remove_set_reserved(&self, who: ScPeerId, protocol: Cow<'static, str>);
+
+    // The PeerId of this node.
+    fn peer_id(&self) -> ScPeerId;
 }
```

Comment:
No idea actually, I wrote this part over a week ago and apparently changing these this way seemed simpler to me then. Changed back, although there are quite a couple more `into`s now everywhere.

I think it might have been due to `PeerId` having to be `pub`?

Reply:
It seems that this introduces `into`s only in the implementation of the trait, as everywhere else we consequently use `PeerId`

Reply:
Yup, there was some fun with the compiler, but now it seems mostly fine.